src/content/docs/en/reference/api-reference.mdx

```diff
@@ -2075,6 +2075,357 @@ export const onRequest = defineMiddleware(async (context, next) => {
+
+## View Transition Events
+
+<p><Since v="3.6.0" /></p>
+
+### `astro:before-preparation` event
+
+An event dispatched at the beginning of a navigation using View Transitions. This event happens before any request is made and any browser state is changed.
+
+This event has the attributes:
+- [`info`](#info)
+- [`state`](#state-option)
+- [`sourceElement`](#sourceelement-option)
+- [`navigationType`](#navigationtype)
+- [`direction`](#direction)
+- [`from`](#from)
+- [`to`](#to)
+- [`formData`](#formdata)
+- [`loader`](#loader)
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astrobefore-preparation).
+
+### `astro:after-preparation`
+
+An event dispatched after the next page in a navigation using View Transitions is loaded.
+
+This event has no attributes.
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astroafter-preparation).
+
+### `astro:before-swap`
+
+An event dispatched after the next page is parsed and prepared and linked into a document in preparation for the transition but before any content is swapped between the documents.
+
+This event can't be canceled. Calling `preventDefault()` is a no-op.
+
+This event has the attributes:
+- [`info`](#info)
+- [`state`](#state-option)
+- [`sourceElement`](#sourceelement-option)
+- [`navigationType`](#navigationtype)
+- [`direction`](#direction)
+- [`from`](#from)
+- [`to`](#to)
+- [`viewTransition`](#viewtransition)
+- [`swap`](#swap)
+
+Read more about how to use this event on the [View Transitions guide](/en/guides/view-transitions/#astrobefore-swap).
+
+### `astro:after-swap`
+
+An event dispatched after the contents of the page have been swapped but before the view transition ends.
+
+The history entry has already been updated when this event is triggered.
+
+### `astro:page-load`
+
+An event dispatched after a page completes loading, be it from a navigation using View Transition or native to the browser.
+
+When view transitions is enabled in the page, code that would normally execute on `DOMContentLoaded` should be changed to execute on this event.
+
+### View Transitions events attributes
+
+<p><Since v="3.6.0" /></p>
+
+#### `info`
+
+<p>
+
+**Type:** `URL`
+</p>
+
+Arbitrary data defined during navigation.
+
+This is the literal value passed on the [`info` option](#info-option) of the [`navigate()` function](#navigate).
+
+#### `sourceElement`
+
+<p>
+
+**Type:** `Element | undefined`
+</p>
+
+The element that triggered the navigation. This can be, for example, an `<a>` element that was clicked.
+
+When using the [`navigate()` function](#navigate), this will be the element specified in the call.
+
+#### `newDocument`
+
+<p>
+
+**Type:** `Document`
+</p>
+
+The document for the next page in the navigation. The contents of this document will be swapped in place of the contents of the current document.
+
+#### `navigationType`
+
+<p>
+  *Type:* `'push' | 'replace' | 'traverse'`
```

Comment:
Same as the last review:
* the new line avoids a possible bug due to markdown used in HTML tags
* Type should be bold, not italic
```suggestion

**Type:** `'push' | 'replace' | 'traverse'`
```